DirectProgramming/DPC++FPGA/Tutorials/GettingStarted/fast_recompile/README.md

```diff
@@ -40,15 +40,15 @@ dpcpp <files.cpp> -o out.fpga -reuse-exe=out.fpga -Xshardware -fintelfpga
 ```
 If `out.fpga` does not exist, `-reuse-exe` is ignored and the FPGA device image is regenerated. This will always be the case the first time a project is compiled.
 
-If `out.fpga` is found, the compiler checks whether any changes affecting the FPGA device code have been made since the last compilation. If no such changes are detected, the compiler reuses the existing FPGA binary and only the host code is recompiled. The recompilation process takes a few minutes. Note that the device code is partially re-compiled (similar to a report flow compile) in order to check that the FPGA binary can safely be reused.
+If `out.fpga` is found, the compiler checks whether any changes affecting the FPGA device code have been made since the last compilation. If no such changes are detected, the compiler reuses the existing FPGA binary, and only the host code is recompiled. The recompilation process takes a few minutes. Note that the device code is partially re-compiled (similar to a report flow compile) to check that the FPGA binary can safely be reused.
 
-If `out.fpga` is found but the compiler is unable to prove that the FPGA device code will yield a result identical to the last compilation, a warning is printed and the FPGA device code is fully recompiled. Since the compiler checks must be conservative, spurious recompilations can sometimes occur when using `-reuse-exe`.
+If `out.fpga` is found but the compiler cannot prove that the FPGA device code will yield a result identical to the last compilation, a warning is printed, and the FPGA device code is fully recompiled. Since the compiler checks must be conservative, spurious recompilations can sometimes occur when using `-reuse-exe`.
 
 ### Using the device link method
 
-The program accompanying this tutorial is separated into two files, `host.cpp` and `kernel.cpp`. Only the `kernel.cpp` file contains device code. 
+The program accompanying this tutorial is separated into two files, `host. cpp` and `kernel. cpp`. Only the `kernel. cpp` file contains device code. 
```

Comment:
While on the whole Grammarly has done a good job, it has done some really stupid things too. I've had to fix many. This one is particularly irksome and bears special mention.